lib/puppet/provider/templates/invoke_dsc_resource.ps1.erb

```diff
@@ -31,13 +34,25 @@ $invokeParams = @{
 <% end -%>
 }
 
-$result = Invoke-DscResource @invokeParams
+try{
+    $result = Invoke-DscResource @invokeParams
+}catch{
+  $exc = $_
+  $data = [PSCustomObject]@{
+    "error" = "execfailure"
+    "exception" = $_.Exception.Message
```

Comment:
I didn't know how far we wanted to go with parsing the exception, as line numbers and context won't be as applicable since it's auto-generated in memory